thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/datalayer/util/StringUtils.java

```diff
@@ -53,7 +53,13 @@ public static Properties decodeCompactedProperties(String propStr) {
     Properties props = new Properties();
     for (String part : SEMICOLON_SPLITTER.split(propStr)) {
       List<String> kvPair = EQUALS_SPLITTER.splitToList(part);
-      props.setProperty(kvPair.get(0), kvPair.get(1));
+      if (kvPair.size() == 2) {
+        props.setProperty(kvPair.get(0), kvPair.get(1));
```

Comment:
Can you trim the result before inserting it into the map? That way we can make sure white-spaces are also cleaned up.

Reply:
Added.